tests/lib/timespec_util/src/main.c

```diff
@@ -329,54 +423,138 @@ ZTEST(timeutil_api, test_timespec_from_timeout)
+			zexpect_equal(actual.ticks, K_TICK_MAX,
 				      "%d: {%" PRId64 "} and {%" PRId64
 				      "} are unexpectedly different",
-				      i, (int64_t)actual.ticks, (int64_t)K_TICKS_FOREVER);
+				      i, (int64_t)actual.ticks, (int64_t)K_TICK_MAX);
+
+			/* workaround */
+			timespec_from_timeout(tspec->timeout, &tick_ts);
+			timespec_add(&tick_ts, &rem);
+			zexpect_true(timespec_equal(&tick_ts, &workaround),
+				     "%d: {%ld, %ld} and {%ld, %ld} are unexpectedly different", i,
+				     tick_ts.tv_sec, tick_ts.tv_nsec, workaround.tv_sec,
+				     workaround.tv_nsec);
 			continue;
 		}
+
+		timespec_from_timeout(tspec->timeout, &tick_ts);
+		timespec_add(&tick_ts, &rem);
+		zexpect_true(timespec_equal(&tick_ts, &tspec->tspec),
+			     "%d: {%ld, %ld} and {%ld, %ld} are unexpectedly different", i,
+			     tick_ts.tv_sec, tick_ts.tv_nsec, tspec->tspec.tv_sec,
+			     tspec->tspec.tv_nsec);
+	}
+
+	/* Special case highlighed in #92709 */
+#if defined(CONFIG_TIMEOUT_64BIT) && (CONFIG_SYS_CLOCK_TICKS_PER_SEC == 100)
+	{
+		struct timespec rem = {};
+		k_timeout_t to = K_TICKS(9223372036854775807LL);
+		struct timespec ts = K_TIMESPEC(92233720368547758LL, 70000000L);
+
+		zexpect_true(K_TIMEOUT_EQ(timespec_to_timeout_rem(&ts, &rem), to),
+			     "timespec_to_timeout(%lld, %lld) != %lld", (long long)ts.tv_sec,
+			     (long long)ts.tv_nsec, (long long)to.ticks);
+		zexpect_true(timespec_equal(&rem, &K_TS_NO_WAIT), "non-zero remainder {%lld, %lld}",
+			     (long long)rem.tv_sec, (long long)rem.tv_nsec);
+
+		TC_PRINT("timespec_to_timeout_rem():\nts: {%lld, %lld} => to: {%" PRId64
+			 "}, rem: {%lld, %lld}\n",
+			 (long long)ts.tv_sec, (long long)ts.tv_nsec, (long long)to.ticks,
+			 (long long)rem.tv_sec, (long long)rem.tv_nsec);
 	}
+#endif
 }
 
 static void *setup(void)
 {
-	printk("CONFIG_TIMEOUT_64BIT=%c\n", CONFIG_TIMEOUT_64BIT ? 'y' : 'n');
-	printk("K_TICK_MAX: %lld\n", (long long)K_TICK_MAX);
-	printk("minimum timeout: {%lld, %lld}\n", (long long)k_timeout_limits[0].tv_sec,
-	       (long long)k_timeout_limits[0].tv_nsec);
-	printk("maximum timeout: {%lld, %lld}\n", (long long)k_timeout_limits[1].tv_sec,
-	       (long long)k_timeout_limits[1].tv_nsec);
+	TC_PRINT("CONFIG_SYS_CLOCK_TICKS_PER_SEC=%d\n", CONFIG_SYS_CLOCK_TICKS_PER_SEC);
+	TC_PRINT("CONFIG_TIMEOUT_64BIT=%c\n", IS_ENABLED(CONFIG_TIMEOUT_64BIT) ? 'y' : 'n');
+	TC_PRINT("K_TICK_MIN: %lld\n", (long long)K_TICK_MIN);
+	TC_PRINT("K_TICK_MAX: %lld\n", (long long)K_TICK_MAX);
+	TC_PRINT("K_TS_MIN: {%lld, %lld}\n", (long long)K_TS_MIN.tv_sec,
+		 (long long)K_TS_MIN.tv_nsec);
+	TC_PRINT("K_TS_MAX: {%lld, %lld}\n", (long long)K_TS_MAX.tv_sec,
+		 (long long)K_TS_MAX.tv_nsec);
+	TC_PRINT("SYS_TIME_T_MIN: %lld\n", (long long)SYS_TIME_T_MIN);
+	TC_PRINT("SYS_TIME_T_MAX: %lld\n", (long long)SYS_TIME_T_MAX);
+	PRINT_LINE;
+
+	zassert_equal(K_TICK_MAX,
+		      IS_ENABLED(CONFIG_TIMEOUT_64BIT) ? 9223372036854775807LL : 4294967294);
```

Comment:
```suggestion
		      IS_ENABLED(CONFIG_TIMEOUT_64BIT) ? INT64_MAX : UINT32_MAX);
```

Reply:
That is incorrect
```cpp
IS_ENABLED(CONFIG_TIMEOUT_64BIT) ? INT64_MAX : UINT32_MAX);
```

This is one of the reasons I would kind of prefer to keep the numerical values there.

I would be happy to add a comment though.

Reply:
> The above is checked symbolically in quite a few places already. I wanted to include a concrete numeric value too almost specifically for this purpose.

We can trust this time_util test that INT64_MAX and UINT32_MAX are what it needs to be..

Reply:
Sure